xla/stream_executor/stream_executor.h

```diff
@@ -324,6 +324,9 @@ class StreamExecutor {
   // Returns the memory limit in bytes supported by this executor.
   virtual int64_t GetMemoryLimitBytes() const = 0;
 
+  // Returns the total number of compute capable devices.
+  virtual int64_t GetDeviceCount() { return 0; };
```

Comment:
Why is this returning 0?

Reply:
since this is the base class for all back-end executors, i gave it the default behavior to return 0 since no back-end library is available at this point. we can also change this to -1 for an invalid value.

Reply:
yea , i notice that it returns 0 when the query of device count fails, so it makes sense to not return 0 here.